themes/src/main/resources/theme/base/login/resources/js/passwordEntropy.js

```diff
@@ -0,0 +1,50 @@
+const clamp = (number, min, max) => {
+    return Math.max(min, Math.min(number, max));
+}
+
+const calculateEntropy = (password) => {
+    let charSetSize = 0;
+    if (/[a-z]/.test(password)) charSetSize += 26;
+    if (/[A-Z]/.test(password)) charSetSize += 26;
+    if (/[0-9]/.test(password)) charSetSize += 10;
+    if (/\W|_/.test(password)) charSetSize += 32; // 149_878
+    return Math.log2(charSetSize ** password.length);
+}
+
+const checkEntropy = (password) =>  {
+    const entropyContainer = document.getElementById("password-entropy");
```

Comment:
This can be moved outside of the function, as the DOM should be available and there is no need to re-query the element every time.